### PR TITLE
3.g et/api/categories

### PR DESCRIPTION
**Pull request by aiidanm**
3. Completed task 3 which involved installing Supertest, express. Setting up my controllers, models and app testing files. 

also resolved a merge conflict on package.json file.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -14,6 +14,25 @@ afterAll(() => {
...
+            expect(Array.isArray(response.body)).toBe(true)
+        })
+    });
+    it('each item in the returned array should contain a key of slug and description ', () => {
+        return request(app)
+            .get("/api/categories")
+            .expect(200)
+            .then((response) => {
+                const arr = response.body
+                arr.forEach((category) => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:39]: Check your array length, we want to check the length is what we expect, there is a jest matcher for this.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -14,6 +14,25 @@ afterAll(() => {
...
+                    expect(category).toMatchObject({
+                        slug: expect.any(String),
+                        description: expect.any(String),
+                    })
+                })
+            })
     });
 });
```
Comment (JVal-NC) [inline __tests__/app.test.js:47]: Just a heads up we will want an error handler for any non existent endpoints. Not needed for this PR but will be needed at some point.  Will be covered in tomorrow's lecture.

File: controllers/categoryControllers.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
-exports.fetchallCategories = () => {
-    
+const {getAllCategories} = require("../models/categoryModels")
+
+exports.fetchallCategories = (req, res, next) => {
+    getAllCategories()
```
Comment (JVal-NC) [inline controllers/categoryControllers.js:4]: Will want a catch block on each controller to catch any unexpected errors.

File: controllers/categoryControllers.js [hunk 4]
```diff
@@ -1,3 +1,8 @@
...
+        .then((categories) => {
+            res.status(200).send(categories.rows)
```
Comment (JVal-NC) [inline controllers/categoryControllers.js]: We will want to send our data on an appropriately named key.

File: app.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 const express = require ('express')
-const {fetchallCategories} = require()
+const {fetchallCategories} = require("./controllers/categoryControllers")
 
 const app = express()
 
+app.get("/api/categories", fetchallCategories)
 
+module.exports = app
```
Comment (JVal-NC) [inline app.js:12]: An error handler for any 500 errors could be useful at this stage to tie in with controller catch block.

**Review by JVal-NC — CHANGES_REQUESTED** [review]
Good start, see comments for changes required.

**Review by Augs0 — APPROVED** [review]
Great job handling the feedback, Aidan. Happy for you to continue

**Pull request merged**